crates/typst/src/layout/grid/layout.rs

```diff
@@ -270,6 +300,23 @@ impl CellGrid {
             };
             let x = resolved_index % c;
             let y = resolved_index / c;
+            let colspan = cell.colspan(styles).get();
+
+            if colspan > c - x {
```

Comment:
Well, at least mathematically speaking this is safer, since `x = (something) % c` and thus `x` is in `0..c`, compared to `colspan` which comes from user input, and thus (at least theoretically) `x + colspan` _could_ result in overflow. Of course that's very unlikely unless you have a supercomputer for all of the columns that'd generate, but it's worth considering I think.

Alternatively, we could move this check to below the `checked_add` for `largest_index`, though that feels a bit odd.

Reply:
Okay, fair enough. I tend not to worry too much about usize overflow when all numbers are typically small, whereas underflow can also happen with small numbers. But what you say is true.